archive/frames/views.py

```diff
@@ -157,31 +158,46 @@ def aggregate(self, request):
         Aggregate field values based on start/end time.
         Returns the unique values shared across all FITS files for site, telescope, instrument, filter, proposal, and obstype.
         """
-        fields = ('SITEID', 'TELID', 'FILTER', 'INSTRUME', 'OBSTYPE', 'PROPID')
+        # TODO: This should be removed after a while, it is just here for temporary API compatibility
+        FIELD_MAPPING = {
+            'SITEID': 'site_id',
+            'TELID': 'telescope_id',
+            'FILTER': 'primary_filter',
+            'INSTRUME': 'instrument_id',
+            'OBSTYPE': 'configuration_type',
+            'PROPID': 'proposal_id'
+        }
+        fields = ('site_id', 'telescope_id', 'primary_filter', 'instrument_id', 'configuration_type', 'proposal_id')
         aggregate_field = request.GET.get('aggregate_field', 'ALL')
+        if aggregate_field in FIELD_MAPPING:
+            aggregate_field = FIELD_MAPPING[aggregate_field]
         if aggregate_field != 'ALL' and aggregate_field not in fields:
             return Response(
                 'Invalid aggregate_field. Valid fields are {}'.format(', '.join(fields)),
                 status=status.HTTP_400_BAD_REQUEST
             )
         query_filters = {}
+        for k in FIELD_MAPPING.keys():
+            if k in request.GET:
+                query_filters[FIELD_MAPPING[k]] = request.GET[k]
         for k in fields:
-            if request.GET.get(k):
+            if k in request.GET:
                 query_filters[k] = request.GET[k]
         if 'start' in request.GET:
-            query_filters['DATE_OBS__gte'] = parse(request.GET['start']).replace(tzinfo=UTC, second=0, microsecond=0)
+            query_filters['observation_date__gte'] = parse(request.GET['start']).replace(tzinfo=UTC, second=0, microsecond=0)
         if 'end' in request.GET:
-            query_filters['DATE_OBS__lte'] = parse(request.GET['end']).replace(tzinfo=UTC, second=0, microsecond=0)
+            query_filters['observation_date__lte'] = parse(request.GET['end']).replace(tzinfo=UTC, second=0, microsecond=0)
         cache_hash = blake2s(repr(frozenset(list(query_filters.items()) + [aggregate_field])).encode()).hexdigest()
         response_dict = cache.get(cache_hash)
         if not response_dict:
-            qs = Frame.objects.order_by().filter(**query_filters)
-            sites = self._get_aggregate_values(qs, 'SITEID', aggregate_field)
-            telescopes = self._get_aggregate_values(qs, 'TELID', aggregate_field)
-            filters = self._get_aggregate_values(qs, 'FILTER', aggregate_field)
-            instruments = self._get_aggregate_values(qs, 'INSTRUME', aggregate_field)
-            obstypes = self._get_aggregate_values(qs, 'OBSTYPE', aggregate_field)
-            proposals = self._get_aggregate_values(qs.filter(L1PUBDAT__lte=timezone.now()), 'PROPID', aggregate_field)
+            qs = Frame.objects.all()
+            qs = DjangoFilterBackend().filter_queryset(request, qs, view=self)
```

Comment:
How does this use `query_filters` to filter down the results?

Reply:
it doesnt, it uses the raw params from the the request that is passed in. query_filters is just used to create a `cache_hash` and remove the seconds/microseconds from the time queries.